internal/hcs/system.go

```diff
@@ -538,12 +538,13 @@ func (computeSystem *System) CreateProcess(c interface{}) (_ *Process, err error
 		Debug("HCS ComputeSystem CreateProcess PID")
 
 	process := newProcess(processHandle, int(processInfo.ProcessId), computeSystem)
-	process.cachedPipes = &cachedPipes{
-		stdIn:  processInfo.StdInput,
-		stdOut: processInfo.StdOutput,
-		stdErr: processInfo.StdError,
+	pipes, err := makeOpenFiles([]syscall.Handle{processInfo.StdInput, processInfo.StdOutput, processInfo.StdError})
+	if err != nil {
+		return nil, makeSystemError(computeSystem, "CreateProcess", "", err, nil)
```

Comment:
There should be a defer `hcsProcessClose` on the process handle on these error paths after creation right?

Reply:
Yes, I think so.

Reply:
Of course, in this case we have leaked the process in the container. I wonder if we should kill it, too?

Reply:
Oh interesting. I guess HCS_PROCESS is not `terminateOnLastHandleClosed` ?

Reply:
We can fix this in another PR if you like.